version: "3"
services:
  plex:
    image: linuxserver/plex
    devices:
      - /dev/dri:/dev/dri
    environment:
      - TZ=${PLEX_TIMEZONE}
      - PLEX_CLAIM=${PLEX_CLAIM}
      - PUID=${MEDIA_LIBRARY_OWNER_USER_ID}
      - PGID=${MEDIA_LIBRARY_OWNER_GROUP_ID}
      - VERSION=docker
    networks:
      - traefik
    volumes:
      - ${PLEX_LIBRARY_LOCATION}:/config
      - ${PLEX_MEDIA_LOCATION}:/data
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      labels:
        - traefik.enable=true
        - traefik.http.routers.plex.rule=Host(`${PLEX_DOMAIN}`)
        - traefik.http.routers.plex.entryPoints=https
        - traefik.http.routers.plex.tls=true
        - traefik.http.routers.plex.tls.certresolver=letsencrypt
        - traefik.http.services.plex.loadbalancer.server.port=32400

networks:
  traefik:
    external: true
